beacon-chain/p2p/broadcaster.go

```diff
@@ -64,11 +57,40 @@ func (s *Service) Broadcast(ctx context.Context, msg proto.Message) error {
 	return nil
 }
 
-const attestationSubnetTopicFormat = "/eth2/%x/committee_index%d_beacon_attestation"
+// BroadcastAttestation broadcasts an attestation to the p2p network.
+func (s *Service) BroadcastAttestation(ctx context.Context, subnet uint64, att *eth.Attestation) error {
+	ctx, span := trace.StartSpan(ctx, "p2p.Broadcast")
+	defer span.End()
+	forkDigest, err := s.forkDigest()
+	if err != nil {
+		return err
+	}
+	topic := attestationToTopic(subnet, forkDigest)
+	span.AddAttributes(trace.StringAttribute("topic", topic))
 
-func attestationToTopic(att *eth.Attestation, forkDigest [4]byte) string {
-	if att == nil || att.Data == nil {
-		return ""
+	buf := new(bytes.Buffer)
+	if _, err := s.Encoding().EncodeGossip(buf, att); err != nil {
+		err := errors.Wrap(err, "could not encode message")
+		traceutil.AnnotateError(span, err)
+		return err
 	}
-	return fmt.Sprintf(attestationSubnetTopicFormat, forkDigest, att.Data.CommitteeIndex)
+
+	if span.IsRecordingEvents() {
+		id := hashutil.FastSum64(buf.Bytes())
+		messageLen := int64(buf.Len())
+		span.AddMessageSendEvent(int64(id), messageLen /*uncompressed*/, messageLen /*compressed*/)
+	}
+
+	if err := s.pubsub.Publish(topic+s.Encoding().ProtocolSuffix(), buf.Bytes()); err != nil {
+		err := errors.Wrap(err, "could not publish message")
+		traceutil.AnnotateError(span, err)
+		return err
+	}
+	return nil
+}
+
+const attestationSubnetTopicFormat = "/eth2/%x/beacon_attestation_%d"
```

Comment:
I feel like this should be near top of the file for easier finding

Reply:
handling this